core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -472,6 +473,7 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
           Selectable.USE_DEFAULT_BUFFER_SIZE,
           Selectable.USE_DEFAULT_BUFFER_SIZE,
           config.requestTimeoutMs,
+          ClientDnsLookup.USE_ALL_DNS_IPS,
```

Comment:
nit: cannot we add a new constructor with the default set to use all instead of changing everywhere? i.e. keep the same constructor but inside it call a new NetworkClient(..., USE_ALL_DNS_IPS, ...)?